src/Umbraco.Web/Routing/IUrlProvider.cs

```diff
@@ -24,19 +23,18 @@ public interface IUrlProvider
         /// when no culture is specified, the current culture.</para>
         /// <para>If the provider is unable to provide a url, it should return <c>null</c>.</para>
         /// </remarks>
-        UrlInfo GetUrl(IUmbracoContext umbracoContext, IPublishedContent content, UrlMode mode, string culture, Uri current);
+        UrlInfo GetUrl(IPublishedContent content, UrlMode mode, string culture, Uri current);
```

Comment:
Seems like by changing this signature we've invented new circular dependencies and same with the overload?

Reply:
Not directly this - The cycle is between these file:
```
DefaultMediaUrlProvider
- PropertyEditorCollection
-- DataEditorCollection
--- GridPropertyEditor
---- HtmlImageSourceParser
----- UrlProvider
------ MediaUrlProviderCollection
------- DefaultMediaUrlProvider
```